### ISSUE TITLE
[Bug] Zenoh does not build on FreeBSD

### ISSUE THREAD

**Issue by gabrik** [eclipse-zenoh/zenoh#302]
### Describe the bug

Neither router nor the examples compile in FreeBSD.
It seems related to the dependency `quinn-udp`, supporting the **QUIC** protocol.

Disabling the features does not solve the issue, the library is still compiled, causing the build to fail.

Here is the complete error:
```
$ cargo build -p zenohd --no-default-features --features transport_tcp
....
Compiling quinn-udp v0.1.2
error[E0422]: cannot find struct, variant or union type `in_pktinfo` in crate `libc`
   --> /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-udp-0.1.2/src/unix.rs:456:41
    |
456 |                       let pktinfo = libc::in_pktinfo {
    |                                           ^^^^^^^^^^ help: a struct with a similar name exists: `in6_pktinfo`
    |
   ::: /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.126/src/unix/bsd/freebsdlike/mod.rs:83:1
    |
83  | / s! {
84  | |     pub struct in_addr {
85  | |         pub s_addr: ::in_addr_t,
86  | |     }
...   |
374 | |     }
375 | | }
    | |_- similarly named struct `in6_pktinfo` defined here

error[E0425]: cannot find value `IP_PKTINFO` in crate `libc`
   --> /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-udp-0.1.2/src/unix.rs:463:58
    |
463 |                     encoder.push(libc::IPPROTO_IP, libc::IP_PKTINFO, pktinfo);
    |                                                          ^^^^^^^^^^ help: a constant with a similar name exists: `IPV6_PKTINFO`
    |
   ::: /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.126/src/unix/bsd/freebsdlike/mod.rs:968:1
    |
968 | pub const IPV6_PKTINFO: ::c_int = 46;
    | ------------------------------------- similarly named constant `IPV6_PKTINFO` defined here

error[E0531]: cannot find unit struct, unit variant or constant `IP_PKTINFO` in crate `libc`
   --> /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-udp-0.1.2/src/unix.rs:523:38
    |
523 |             (libc::IPPROTO_IP, libc::IP_PKTINFO) => unsafe {
    |                                      ^^^^^^^^^^ help: a constant with a similar name exists: `IPV6_PKTINFO`
    |
   ::: /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.126/src/unix/bsd/freebsdlike/mod.rs:968:1
    |
968 | pub const IPV6_PKTINFO: ::c_int = 46;
    | ------------------------------------- similarly named constant `IPV6_PKTINFO` defined here

error[E0412]: cannot find type `in_pktinfo` in crate `libc`
   --> /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-udp-0.1.2/src/unix.rs:524:52
    |
524 |                   let pktinfo = cmsg::decode::<libc::in_pktinfo>(cmsg);
    |                                                      ^^^^^^^^^^ help: a struct with a similar name exists: `in6_pktinfo`
    |
   ::: /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.126/src/unix/bsd/freebsdlike/mod.rs:83:1
    |
83  | / s! {
84  | |     pub struct in_addr {
85  | |         pub s_addr: ::in_addr_t,
86  | |     }
...   |
374 | |     }
375 | | }
    | |_- similarly named struct `in6_pktinfo` defined here

error[E0308]: mismatched types
    --> /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-udp-0.1.2/src/unix.rs:250:72
     |
250  |             unsafe { libc::sendmmsg(io.as_raw_fd(), msgs.as_mut_ptr(), num_transmits as u32, 0) };
     |                      --------------                                    ^^^^^^^^^^^^^^^^^^^^ expected `usize`, found `u32`
     |                      |
     |                      arguments to this function are incorrect
     |
note: function defined here
    --> /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.126/src/unix/bsd/freebsdlike/freebsd/mod.rs:4130:12
     |
4130 |     pub fn sendmmsg(
     |            ^^^^^^^^
help: you can convert a `u32` to a `usize` and panic if the converted value doesn't fit
     |
250  |             unsafe { libc::sendmmsg(io.as_raw_fd(), msgs.as_mut_ptr(), (num_transmits as u32).try_into().unwrap(), 0) };
     |                                                                        +                    +++++++++++++++++++++

error[E0308]: mismatched types
    --> /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/quinn-udp-0.1.2/src/unix.rs:359:17
     |
356  |             libc::recvmmsg(
     |             -------------- arguments to this function are incorrect
...
359  |                 bufs.len().min(BATCH_SIZE) as libc::c_uint,
     |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected `usize`, found `u32`
     |
note: function defined here
    --> /home/ato/.cargo/registry/src/github.com-1ecc6299db9ec823/libc-0.2.126/src/unix/bsd/freebsdlike/freebsd/mod.rs:4136:12
     |
4136 |     pub fn recvmmsg(
     |            ^^^^^^^^
help: you can convert a `u32` to a `usize` and panic if the converted value doesn't fit
     |
359  |                 (bufs.len().min(BATCH_SIZE) as libc::c_uint).try_into().unwrap(),
     |                 +                                          +++++++++++++++++++++

Some errors have detailed explanations: E0308, E0412, E0422, E0425, E0531.
For more information about an error, try `rustc --explain E0308`.
error: could not compile `quinn-udp` due to 6 previous errors
warning: build failed, waiting for other jobs to finish
```

### To reproduce

Clone zenoh
Build zenoh with `cargo build -p zenohd --no-default-features --features transport_tcp`

### System info

- FreeBSD 13.1 x86_64
- Intel(R) Core(TM) i7-8550U CPU @ 1.80GHz
- rustc 1.63.0 (4b91a6ea7 2022-08-08)
- FreeBSD clang version 13.0.0 (git@github.com:llvm/llvm-project.git llvmorg-13.0.0-0-gd7b669b3a303)

**Comment by JEnoch** [comment]
The root of this issue is `quinn-udp` failing to build on FreeBSD, as reported in quinn-rs/quinn#1142.

Waiting this to be fixed, #303 allows to build on FreeBSD all zenoh artifacts, disabling QUIC transport.
For such, use this command:
```bash
cargo build -p zenohd -p zenoh-examples -p zenoh-ext -p zenoh-plugin-rest -p zenoh-plugin-storage-manager --lib --bins --examples --no-default-features --features zenoh/auth_pubkey,zenoh/auth_usrpwd,zenoh/shared-memory,zenoh/transport_tcp,zenoh/transport_tls,zenoh/transport_udp,zenoh/transport_unixsock-stream
```
This command builds only the packages producing the zenoh binaries (to avoid building `zenoh-link-quic` package), with all `zenoh` default features except `transport-quic`.

**Comment by Mallets** [comment]
Closing the issue because https://github.com/eclipse-zenoh/zenoh/pull/303 has been merged.
